src/libs/Environment/Environment.js

```diff
@@ -43,7 +44,12 @@ function isInternalTestBuild() {
  */
 function getEnvironmentURL() {
     return new Promise((resolve) => {
-        getEnvironment().then((environment) => resolve(ENVIRONMENT_URLS[environment]));
+        getEnvironment().then((environment) => 
+        {
+            // Our approach is to use staging url only for staging environment and use production url for all other environments.
+            const environmentURL = (environment === CONST.ENVIRONMENT.STAGING) ?  Url.addTrailingForwardSlash(CONST.STAGING_NEW_EXPENSIFY_URL) : CONST.NEW_EXPENSIFY_URL;
+            resolve(environmentURL);
+        });
```

Comment:
As mentioned in the details section, 'Copy Link' is the only other place where we use getEnvironmentURL(). There, we need this change as the use case is similar to the `share code` one.

Reply:
Oh! Didn't notice the extra slash. Thanks for pointing it out. Will remove the extra slash [here](https://github.com/Expensify/App/blob/main/src/pages/home/report/ContextMenu/ContextMenuActions.js#L209)

Reply:
As mentioned above, why not include a slash in the code that uses this function? Ideally we can keep that code the same and make the least amount of code changes.

Reply:
If I am not mistaken, wouldn't it be easier to not modify this file at all? and use `Url.addTrailingForwardSlash` in the file where we are actually using the URL?

Reply:
Actually, our existing approach is to ensure that the configured URLs have a trailing slash using the helper function `Url.addTrailingForwardSlash`. If the configured URL already has a trailing slash, it will ignore. Else, it will add the trailing slash. If we want to follow the approach of ensuring that there is no trailing slash in the configured URLs, we will have to implement additional code to remove the trailing slash from configured URLs. Also, the URLs in the environment configuration files have a trailing slash. So, I strongly think that we should not attempt a change of approach here towards handling of trailing slash in URLs.

Reply:
@AndrewGable @mananjadhav Please let me know your comments on above too so that we can decide and take it further

Reply:
@rojiphil I've commented again. The reason I think we went ahead with environment was it takes care of DEV env too.